src/components/FinishRecipeButton/index.js

```diff
@@ -29,17 +29,26 @@ class FinishRecipeButton extends React.Component {
     };
 
     setToStorage('doneRecipes', [...recipesDone, newDoneRecipe]);
-
+    this.removeStepFromStorage(id);
     history.push('/receitas-feitas');
   }
 
+  removeStepFromStorage(id) {
+    const inProgressRecipes = getFromStorage('inProgressRecipes') || [];
+    delete inProgressRecipes[id];
+    setToStorage('inProgressRecipes', inProgressRecipes);
+    if (Object.keys(inProgressRecipes).length === 0) {
+      localStorage.removeItem('inProgressRecipes');
+    }
+  }
+
   render() {
     const { id, ingredients, inProgress } = this.props;
 
     return (
       <button
         type="button"
-        onClick={ this.handleRecipeDone }
+        onClick={ () => this.handleRecipeDone() }
```

Comment:
Acredito que não seja necessário o uso de `arrow function` nesse evento, o que acha?
```suggestion
        onClick={ this.handleRecipeDone }
```

Reply:
Opa, pode ser.